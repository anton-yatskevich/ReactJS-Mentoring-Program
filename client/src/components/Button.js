import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
    onClickHandler, textValue, value, searchField
}) => (
    <button
        type="button"
        onClick={() => onClickHandler(value)}
        className={searchField === value ? 'button--active' : ''}
    >
        {textValue}
    </button>
);

Button.propTypes = {
    onClickHandler: PropTypes.func,
    textValue: PropTypes.string,
    value: PropTypes.string,
    searchField: PropTypes.string
};

Button.defaultProps = {
    onClickHandler: null,
    textValue: '',
    value: '',
    searchField: ''
};

export default Button;
