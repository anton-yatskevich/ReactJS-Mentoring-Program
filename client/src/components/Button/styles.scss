.button {
    padding: 5px 20px;
    font-size: 20px;
    text-transform: uppercase;
    background-color: rgb(226, 226, 226);
    border: none;
    color: rgb(148, 0, 0);
    text-decoration: none;
    margin-left: 15px;
    cursor: pointer;
}