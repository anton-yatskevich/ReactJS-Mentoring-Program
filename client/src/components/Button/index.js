import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash.noop';
import classNames from 'classnames';
import './styles.scss';

const Button = ({
    onClickHandler, textValue, value, isActive, className
}) => (
    <button
        type="button"
        onClick={() => onClickHandler(value)}
        className={classNames(className, { [`${className}--active`]: isActive })}
    >
        {textValue}
    </button>
);

Button.propTypes = {
    onClickHandler: PropTypes.func,
    textValue: PropTypes.string,
    value: PropTypes.string,
    isActive: PropTypes.bool,
    className: PropTypes.string
};

Button.defaultProps = {
    onClickHandler: noop,
    textValue: '',
    value: '',
    isActive: false,
    className: 'button'
};

export default Button;
