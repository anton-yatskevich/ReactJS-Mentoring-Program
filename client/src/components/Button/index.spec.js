import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

describe('Button component', () => {
    it('should be rendered with default props', () => {
        const ButtonComponent = shallow(<Button />);
        expect(ButtonComponent).toMatchSnapshot();
    });

    it('should has active class if the button is active', () => {
        const props = {
            isActive: true,
            className: 'btn',
            textValue: 'test'
        };
        const button = shallow(<Button {...props} />).find('button');

        expect(button.hasClass('btn--active')).toEqual(true);
        expect(button.text()).toBe('test');
    });

    it('should call onClick handler when click event fired', () => {
        const props = {
            onClickHandler: jest.fn()
        };
        const ButtonComponent = shallow(<Button {...props} />);

        ButtonComponent.simulate('click');
        expect(props.onClickHandler).toHaveBeenCalled();
    });
});
