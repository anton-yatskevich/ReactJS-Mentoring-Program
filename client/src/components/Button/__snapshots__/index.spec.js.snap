// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should be rendered with default props 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
/>
`;
